testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/deployment/DeploymentScannerOperationRollbackTestCase.java

```diff
@@ -0,0 +1,121 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.test.manualmode.deployment;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.FAILED;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OUTCOME;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.PATH;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.RELATIVE_TO;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SYSTEM_PROPERTY;
+import static org.junit.Assert.assertThat;
+
+import java.io.File;
+
+import javax.inject.Inject;
+import org.jboss.as.controller.PathAddress;
+
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.server.deployment.scanner.FileSystemDeploymentScanHandler;
+import org.jboss.as.test.integration.domain.management.util.DomainTestSupport;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.as.test.shared.TimeoutUtil;
+import org.jboss.dmr.ModelNode;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.ServerControl;
+import org.wildfly.core.testrunner.ServerController;
+import org.wildfly.core.testrunner.WildflyTestRunner;
+
+/**
+ * Testing the rollback of a composite operation with a manual scan.
+ * @author <a href="mailto:ehugonne@redhat.com">Emmanuel Hugonnet</a>  (c) 2015 Red Hat, inc.
+ */
+@RunWith(WildflyTestRunner.class)
+@ServerControl(manual = true)
+public class DeploymentScannerOperationRollbackTestCase extends AbstractDeploymentScannerOperationTestCase {
+
+    @Inject
+    private ServerController container;
+
+    @Test
+    public void testRollBackAfterOperationfailure() throws Exception {
+        container.start();
+        try {
+            try (ModelControllerClient client = TestSuiteEnvironment.getModelControllerClient()) {
+                final File deploymentOne = new File(deployDir, "deployment-one.jar");
+                createDeployment(deploymentOne, "org.jboss.modules");
+                // Add a new de
+                addDeploymentScanner(client);
+                prepareRollback(client);
+                try {
+                    assertThat(exists(client, DEPLOYMENT_ONE), is(false));
+                    runFailingScan(client);
+                    // Wait until deployed ...
+                    long timeout = System.currentTimeMillis() + TimeoutUtil.adjust(30000);
+                    while (!exists(client, DEPLOYMENT_ONE) && System.currentTimeMillis() < timeout) {
+                        Thread.sleep(100);
+                    }
```

Comment:
This will always take 30 secs to prove a scan didn't happen. I don't have any clever ideas for proving a negative, but how about 2 or 3 secs?